packages/compiler/core/checker.ts

```diff
@@ -2265,15 +2266,19 @@ export function createChecker(program: Program): Checker {
       return;
     }
 
-    if (inheritedPropertyNames?.has(newProp.name)) {
-      program.reportDiagnostic(
-        createDiagnostic({
-          code: "override-property",
-          format: { propName: newProp.name },
-          target: diagnosticTarget ?? newProp,
-        })
-      );
-
+    if (inheritedProperties?.has(newProp.name)) {
+      const inheritedProp = inheritedProperties.get(newProp.name)!;
+      const [isAssignable, _] = isTypeAssignableTo(newProp.type, inheritedProp.type, newProp);
+      const parentIntrinsic = isIntrinsic(program, inheritedProp.type);
+      if (!parentIntrinsic || !isAssignable) {
+        program.reportDiagnostic(
+          createDiagnostic({
+            code: "override-property",
+            format: { propName: newProp.name },
```

Comment:
I think this diagnostic needs to be updated given that you can override a property with this PR (it just needs to be a subtype of the parent type).